src/renderer.ts

```diff
@@ -156,6 +158,632 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+            height: (d: any) => (d < threshold) ? (barheight / 2 - scaleheight(context.rowHeight(i), d)) : scaleheight(context.rowHeight(i), d),
+            fill: color
+          });
+        rect.exit().remove();
+      }
+    };
+  }
+
+  createCanvas(col: Column, context: ICanvasRenderContext): ICanvasCellRenderer {
+
+
+    const bins = (<any> col.desc).datalength;
+    const min = (<any> col.desc).domain[0];
+    const max = (<any> col.desc).domain[1];
+
+    const colorrange = (<any> col.desc).colorrange;
+    const defaultcolor = ['blue', 'red'];
+
+    const mincolor = (colorrange === undefined || null) ? defaultcolor[0] : colorrange[0];
+    const maxcolor = (colorrange === undefined || null) ? defaultcolor[1] : colorrange[1];
+
+    const celldimension = (col.getWidth() / bins);
+    var threshold = (<any> col.desc).threshold || 0;
+    var barheight = 13;
+    var scale = d3.scale.linear();
+    var color: any = d3.scale.linear<number,string>();
+
+    function scaleheight(barheight, data) {
+      scale = (min < 0) ? (scale.domain([min, max]).range([0, barheight / 2])) : (scale.domain([min, max]).range([0, barheight]));
+      return (scale(data));
+    };
+    color.domain([min, 0, max]).range([mincolor, 'white', maxcolor]);
+
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+      var data = col.getValue(d.v, i);
+      data.forEach(function (d, i) {
+        var xpos = (i === null || 0) ? 0 : (i * celldimension);
+        var ypos;
+        if (min < 0) {
+          ypos = (d < threshold) ? (context.rowHeight(i) / 2) : context.rowHeight(i) / 2 - scaleheight(context.rowHeight(i), d);   // For positive and negative value
+        } else {
+          ypos = context.rowHeight(i) - scaleheight(context.rowHeight(i), d);
+        }
+        var height = (d < threshold) ? (barheight / 2 - scaleheight(context.rowHeight(i), d)) : scaleheight(context.rowHeight(i), d);
+        ctx.fillStyle = color(d);
+        ctx.fillRect(xpos, ypos, celldimension, height);
+      });
+    };
+  }
+
+
+}
+class BoxplotCellRenderer extends DefaultCellRenderer {
+
+  createSVG(col: Column, context: IDOMRenderContext): ISVGCellRenderer {
+
+    function getPercentile(data, percentile) {
+
+      var index = (percentile / 100) * data.length;
+      var result;
+      if (Math.floor(index) === index) {
+        result = (data[(index - 1)] + data[index]) / 2;
+      } else {
+        result = data[Math.floor(index)];
+      }
+      return result;
+    }
+
+    function numSort(a, b) {
+      return a - b;
+    }
+
+    const padding = context.option('rowPadding', 1);
+    const min = (<any> col.desc).domain[0];
+    const max = (<any> col.desc).domain[1];
+    var scale = d3.scale.linear().domain([min, max]).range([0, col.getWidth()]); // Constraint the window width
+
+    return {
+
+      template: `<g class="boxplotcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+
+        var data = col.getValue(d.v, i);
```

Comment:
computing the boxplot values should be extracted to the model class, avoiding to much logic within the render code